lib/OpenQA/Worker/WebUIConnection.pm

```diff
@@ -314,9 +314,9 @@ sub send {
         $msg //= $err->{message};
         if (my $error_code = $err->{code}) {
             $msg = "$error_code response: $msg";
-            if ($error_code == 404) {
-                # don't retry on 404 errors (in this case we can't expect different
-                # results on further attempts)
+            if (400 <= $error_code && $error_code < 500) {
```

Comment:
`$error_code >= 400 && $error_code < 500` is much more readable too me (for some reason).

Reply:
sorry, not convinced. Only if I must